buffer.go

```diff
@@ -334,7 +353,11 @@ func (buf *Buffer) Len() int {
 
 // Less returns true if row[i] < row[j] in the buffer.
 func (buf *Buffer) Less(i, j int) bool {
-	for _, col := range buf.sorted {
+	for _, entry := range buf.sorted {
+		if entry.buffer == nil {
+			continue
+		}
+		col := entry.buffer
```

Comment:
This looks like the fix? Could we just go with a nil check here and drop all the other changes? I don't understand what we need the rest of the modifications for.